post-scan-actions/azure-python-promote-or-quarantine/promoteOrQuarantineFunction/ScanResultHandler/handler.py

```diff
@@ -121,18 +123,31 @@ def main(message: func.ServiceBusMessage):
 
 
 def copy_object(source_blob_url, container, blob_name, metadata, tags, dest_blob_service_client: BlobServiceClient):
-    container_client = dest_blob_service_client.get_container_client(container)
-    if not container_client.exists():
-        container_client.create_container()
-
-    blob_client = container_client.get_blob_client(blob_name)
-    copy_status = blob_client.start_copy_from_url(
-        source_url=source_blob_url,
-        metadata=metadata,
-        tags=tags,
-        requires_sync=True
-    )
-    logging.info(f'copy status: {copy_status}')
+    try:
+        container_client = dest_blob_service_client.get_container_client(container)
+        if not container_client.exists():
+            container_client.create_container()
+        blob_client = container_client.get_blob_client(blob_name)
+        copy_result = blob_client.start_copy_from_url(
+            source_url=source_blob_url,
+            metadata=metadata,
+            tags=tags,
+            requires_sync=False
+        )
+        logging.info(f'copy result: {copy_result}')
+
+        retries = 0
+        copy_status = copy_result['copy_status']
+        while copy_status != 'success' and retries * SLEEP_INTERVAL < TIMEOUT:

```

Comment:
It's much clear to use the retry number as the condition instead using the not accurate execution time.

Reply:
At the moment, the condition retries * SLEEP_INTERVAL < TIMEOUT is never met because the function times out before it can be satisfied. Would you consider removing this condition?

Reply:
This is one of the design decisions that I wasn't sure of. This condition is here because the user can change the function timeout time in host.json. This TIMEOUT was meant to be set as a dummy value just to make sure the while loop doesn't run forever. And I agree setting a hard time limit doesn't seem to be the best choice.

I wouldn't use retry time to limit the while loop because it's not clear how long each retry might take and this might cause the function to prematurely stop before the function timeout.

Reply:
After second thought, I think I will remove the condition, and if supposedly there is something wrong with the copying causing the status to remain in the pending state, we will just let the function timeout.